jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -82,13 +89,26 @@ class JobRunExecutorActor(
     context.become(starting)
   }
 
+  def scheduleTimeout(): Unit = {
+    import scala.concurrent.duration._
+
+    jobRun.startingDeadline.foreach { deadline =>
+      val now = clock.now()
+      val from = run.createdAt.plus(deadline.toMillis)
+      val timeout = Seconds.secondsBetween(now, from).getSeconds.seconds
+      log.info(s"Timeout scheduled for ${jobRun.id} at $timeout")
+      startingDeadlineTimer = Some(context.system.scheduler.scheduleOnce(timeout, self, StartTimeout))
```

Comment:
What do you think about using the scheduler from constructor?

Reply:
Yeah, this should totally use the constructor provided scheduler. Good catch. I think the test would depend on this? (How does the test pass without using the test-provided scheduler?)

Reply:
thanks!